conda_smithy/templates/circle.yml.tmpl

```diff
@@ -9,6 +9,13 @@ general:
 {% endblock -%}
 {% block env_dependencies -%}
 {%- if matrix -%}
+checkout:
+  post:
+    # Checkout the merged PR for testing as CircleCI will just use the PR head otherwise.
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git fetch origin +refs/pull/$CIRCLE_PR_NUMBER/head:pr/$CIRCLE_PR_NUMBER/head ; fi
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git fetch origin +refs/pull/$CIRCLE_PR_NUMBER/merge:pr/$CIRCLE_PR_NUMBER/merge ; fi
+    - if [[ -n "${CIRCLE_PR_NUMBER}" ]]; then git checkout -qf pr/$CIRCLE_PR_NUMBER/merge ; fi
```

Comment:
This is being tracked in issue ( https://github.com/conda-forge/conda-smithy/issues/316 ).